celery/schedules.py

```diff
@@ -101,6 +101,11 @@ def to_local(self, dt):
             return timezone.to_local_fallback(dt)
         return dt
 
+    def __eq__(self, other):
+        if isinstance(other, BaseSchedule):
+            return other.nowfun == self.nowfun
```

Comment:
I'm not too worried about code equality for functions, more so that the nowfun will evaluate equivalent and operate the same after being pickled and unpickled.  It wouldn't work for lambdas.  General functions will pickle and unpickle in a way that preserves equality.

Note that celery beat, on startup, pickles and unpickles all schedules, so before this, nowfun is entirely nonfunctional in general.  I couldn't find a way that a passed in nowfun could be preserved and used after celery beat's startup.